src/EditorFeatures/Core/Shared/Utilities/UIOperationContext.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Utilities;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
+{
+    /// <summary>
+    /// Wrapper around a <see cref="IUIThreadOperationContext"/> adapting it to be an <see cref="IOperationContext"/>.
+    /// </summary>
+    internal class UIOperationContext : IOperationContext
+    {
+        private readonly IUIThreadOperationContext _operationContext;
+
+        public UIOperationContext(IUIThreadOperationContext operationContext)
+        {
+            _operationContext = operationContext;
+        }
+
+        public CancellationToken CancellationToken => _operationContext.UserCancellationToken;
```

Comment:
effectively, as this is a wrapper of a real IUIThreadOperationContext , we just forward everything to it.